Documentation/design-docs/first-class-structs.md

```diff
@@ -90,20 +90,21 @@ encountered by most phases of the JIT:
 * The lhs of a struct assignment is a block node or local
   * `GT_OBJ` nodes represent the “shape” info via a struct handle, along with the GC info
     (location and type of GC references within the struct).
-    * These are used only to represent struct values that contain GC references.
+    * These are currently used only to represent struct values that contain GC references (although see below).
   * `GT_BLK` nodes represent struct types with no GC references, or opaque blocks of fixed size.
     * These have no struct handle, resulting in some pessimization or even incorrect
       code when the appropriate struct handle can't be determined.
     * These never represent lvalues of structs that contain GC references.
-    * Proposed: Like `GT_OBJ`, these would contain a pointer to the `ClassLayout`.
+    * Proposed: When a class handle is available, these would remain as `GT_OBJ` since after 
+      [#21705](https://github.com/dotnet/coreclr/pull/21705) they are no longer large nodes.
   * `GT_STORE_OBJ` and `GT_STORE_BLK` have the same structure as `GT_OBJ` and `GT_BLK`, respectively
     * `Data()` is op2
   * `GT_DYN_BLK` and `GT_STORE_DYN_BLK` (GenTreeDynBlk extends GenTreeBlk)
     * Additional child `gtDynamicSize`
     * Note that these aren't really struct types; they represent dynamically sized blocks
       of arbitrary data.
-  * For `GT_LCL_FLD` nodes, the shape information is obtained via an index into the `ClassLayout` cache.
-  * For `GT_LCL_VAR` nodes, the shape information is obtained via the `LclVarDsc`
+  * For `GT_LCL_FLD` nodes, we don't retain shape information, except indirectly via the `FieldSeqNode`.
```

Comment:
I'll add the type info to LCL_FLD soon enough. I'm just trying to find a reasonable small and safe use case for a start so it doesn't become a huge PR. Though probably this will have to wait for the BLK -> OBJ change. It would be easy to produce struct typed LCL_FLDs from `OBJ(ADDR(LCL_FLD))` trees.